Src/PerformanceCollector/Filtering.Shared/Implementation/Service contract/CalculatedMetricInfo.cs

```diff
@@ -77,7 +77,19 @@ public override string ToString()
                 this.TelemetryType,
                 this.Projection,
                 this.Aggregation,
-                string.Join(", ", (this.FilterGroups ?? new FilterConjunctionGroupInfo[0]).Select(filterGroup => filterGroup.ToString())));
+                this.FilterGroupsToString());
+        }
+
+        private string FilterGroupsToString()
+        {
+            if (this.FilterGroups == null)
+            {
+                return string.Empty;
+            }
+            else
+            {
+                return string.Join(", ", this.FilterGroups.Select(filterGroup => filterGroup.ToString()));
```

Comment:
Do we need a Format Provider for this and other ToString() methods as well?

Reply:
in this instance no..
this is our overridden ToString() method that has a custom string builder. Inside our custom method we use CultureInfo.InvariantCulture.